src/oneiro/pipelines/civitai_checkpoint.py

```diff
@@ -777,24 +782,35 @@ def generate(
 
         # Build generation kwargs
         gen_kwargs: dict[str, Any] = {
-            "prompt": prompt,
             "num_inference_steps": steps,
             "guidance_scale": guidance_scale,
             "generator": generator,
         }
 
-        # Add negative prompt only for pipelines that support it
-        if self._pipeline_config.supports_negative_prompt and negative_prompt:
-            gen_kwargs["negative_prompt"] = negative_prompt
+        # Check if we need long prompt handling (>77 tokens)
+        use_long_prompt = self._should_use_long_prompt(prompt, negative_prompt)
```

Comment:
@dobbyphus, why not always use long prompts? would that not always allow the a1111 weighting?

Reply:
@dobbyphus i dont' buy the reduced overhead, its going to happen either by us or by the model it self, and we get the benefit of having one way to handle weights. Please fix this and simplify it